src/dask_awkward/lib/io/io.py

```diff
@@ -640,8 +642,48 @@ def from_map(
 
     if io_func_implements_report(io_func):
         if cast(ImplementsReport, io_func).return_report:
-            res = result.map_partitions(first, meta=array_meta, output_divisions=1)
-            rep = result.map_partitions(second, meta=empty_typetracer())
+            res = result.map_partitions(
+                first, meta=array_meta, label=label, output_divisions=1
+            )
+
+            concat_fn = partial(
+                ak.concatenate,
+                axis=0,
+            )
+
+            split_every = 8
+
+            rep_trl_label = f"{label}-report"
+            rep_trl_token = tokenize(result, second, concat_fn, split_every)
+            rep_trl_name = f"{rep_trl_label}-{rep_trl_token}"
+            rep_trl_tree_node_name = f"{rep_trl_label}-tree-node-{rep_trl_token}"
+
+            rep_part = result.map_partitions(
+                second, meta=empty_typetracer(), label=f"{label}-partitioned-report"
+            )
+
+            rep_trl = AwkwardTreeReductionLayer(
+                name=rep_trl_name,
+                name_input=rep_part.name,
+                npartitions_input=rep_part.npartitions,
+                concat_func=concat_fn,
+                tree_node_func=lambda x: x,
```

Comment:
We should just define `identity()`. Pretty sure dask has one.

Reply:
I'll get this in a housekeeping PR in just a sec.